### ISSUE TITLE
floating point timestamps rejected 

### ISSUE THREAD

**Issue by enthooz** [prometheus/prometheus#10259]
**What did you do?**

- Use `promtool check metrics` to lint metrics that have one or more floating point timestamps.

Example:

```
# TYPE my_metric gauge
# UNIT my_metric 
# HELP my_metric This is a dummy metric.
my_metric{my_label="xyz"} 456.2 1234567890.123
# EOF
```

**What did you expect to see?**

- Successful linting, i.e. no error message and 0 exit code.
- [OpenMetrics spec](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#numbers) indicates that floating point numbers "[...] MAY be used to encode nanosecond resolution timestamps."

**What did you see instead? Under which circumstances?**

- Error message and non-zero exit code.

**Environment**

* System information:

```
Darwin 20.6.0 x86_64
```

* Prometheus version:

```
prometheus, version 2.32.1 (branch: HEAD, revision: 41f1a8125e664985dd30674e5bdf6b683eff5d32)
  build user:       root@3e78e16c59b8
  build date:       20211217-21:54:37
  go version:       go1.17.5
  platform:         darwin/amd64
```

* Alertmanager version:

n/a

* Prometheus configuration file:

n/a

* Alertmanager configuration file:

n/a

* Logs:

```
$ cat test
# TYPE my_metric gauge
# UNIT my_metric
# HELP my_metric This is a dummy metric.
my_metric{my_label="xyz"} 456.2 1234567890.456
# EOF

$ cat test | prometheus-2.32.1.darwin-amd64/promtool check metrics
error while linting: text format parsing error in line 4: expected integer as timestamp, got "1234567890.456"

$ echo $?
1
```

**Comment by beorn7** [comment]
Yeah, AFAIK `promtool check metrics` expects the classic Prometheus text format and doesn't do OpenMetrics at all.

**Comment by LeviHarrison** [comment]
There's an issue to add OpenMetrics support, https://github.com/prometheus/prometheus/issues/8932, but it's waiting on a new parser to be merged.

**Comment by roidelapluie** [comment]
Thanks for your report. I am closing this as a duplicate of #8932.
